music21/musicxml/m21ToXml.py

```diff
@@ -6552,6 +6617,29 @@ def testFromSoundingPitch(self):
         self.assertEqual(len(root.findall('.//transpose')), 1)
         self.assertEqual(root.find('.//step').text, 'D')
 
+    def testMultipleInstruments(self):
+        from music21 import instrument
+
+        p1 = stream.Part([
+            stream.Measure([instrument.Oboe(), note.Note(type='whole')]),
+            stream.Measure([instrument.Flute(), note.Note(type='whole')]),
+            stream.Measure([instrument.Oboe(), note.Note(type='whole')]),
+        ])
+        p2 = stream.Part([
+            stream.Measure([instrument.Oboe(), note.Note(type='whole')]),
+            stream.Measure([instrument.Flute(), note.Note(type='whole')]),
+            stream.Measure([instrument.Oboe(), note.Note(type='whole')]),
+        ])
+        s = stream.Score([p1, p2])
+        scEx = ScoreExporter(s)
+        tree = scEx.parse()
+        self.assertEqual(len(tree.findall('.//score-instrument')), 4)
```

Comment:
should we be testing which instruments were the 4 that are retained?

Reply:
This code is complex enough and I don't fully understand it, that I'd rather have the instrumentIds be A, B, C, D etc. rather than randomized (unless necessary) in order to be able to visualize what is happening here by looking at the whole Tree (or at least relevant parts of it).

Reply:
Sure, I'll give this another look and make it as easy as possible to understand. This is a score for two woodwind players both doubling on flute and oboe. They both switch to flute and then back to oboe. There are six m21 instruments to represent this, but the `<score-instrument>` tags need just four, since no musicXML `<part>` needs two oboes in it, etc., unless there is a patch change/MIDI instrument change. Which kind of gets to your other question about whether we should deduplicate instruments that differ in MIDI program but resolve to same m21 instrument. I'll have another look.